plugin/src/main/java/com/exadel/aem/toolkit/plugin/utils/NamingUtil.java

```diff
@@ -120,14 +122,18 @@ public static String getUniqueName(String value, String defaultValue, Element co
     }
 
     /**
-     * Retrieves the member name value with the Java getter prefix stripped
-     * @param value String used as the source
+     * Retrieves the member name value with the Java getter prefix stripped for method, otherwise returns a full source
+     * name
+     * @param source Source used to get source name and check source type
      * @return String value
      */
-    public static String stripGetterPrefix(String value) {
-        if (StringUtils.startsWith(value, PREFIX_GET)) {
+    public static String stripGetterPrefix(Source source) {
+        String value = source.getName();
+
+        if (source instanceof MethodSourceImpl && StringUtils.startsWith(value, PREFIX_GET)) {
```

Comment:
https://betterprogramming.pub/code-against-interfaces-not-implementations-37b30e7ab992
I'm afraid that the holy righteous dogs of OOP will punish us for that)
I suggest the following approach
```
if (source.adaptTo(Method.class) != null) { ... }
```